lesson21/src/js/main.js

```diff
@@ -0,0 +1,216 @@
+    };
+
+    table.appendChild(thead).appendChild(tr);
+}
+
+const renderTableBody = usersData => {
+    const tbody = document.createElement("tbody");
+    tbody.id = "js-usersTableBody";
+
+    for(const user of usersData){
+        const fragment = document.createDocumentFragment();
+        const tr = document.createElement("tr");
+
+        Object.keys(usersTableColumn).forEach((column) => {
+            const td = createAttributedElements({
+                tag:"td",
+                valuesByAttributes:{
+                    class:"border border-gray-300 px-4 py-2"
+                },
+                str:user[column]
+            })
+            fragment.appendChild(tr).appendChild(td);
+        });
+
+        table.appendChild(tbody).appendChild(fragment);
+    }
+}
+
+const sortButtonAttributes = [
+    {state:"asc",src:"../img/asc.svg", alt:"asc-image"},
+    {state:"desc",src:"../img/desc.svg",alt:"desc-image"},
+    {state:"both",src:"../img/both.svg",alt:"both-image"}
+]
+
+const sortCategories = ["id"];
+
+const createSortButtons = (columnKey) => {
+    const sortButtonsBox = createAttributedElements ({
+        tag:"div",
+        valuesByAttributes:{
+            id:`js-${columnKey}Buttons-Box`,
+            class: "w-10 inline-block align-middle"
+        }
+    })
+
+    const fragment = document.createDocumentFragment();
+
+    for(const button of sortButtonAttributes){
+        const sortButton = createAttributedElements({
+            tag:"button",
+            valuesByAttributes:{
+                class: `js-${columnKey}SortButton`,
+                'data-state':button.state
+            }
+        });
+
+        button.state !== "both" && sortButton.classList.add("hidden");
+
+        const sortButtonImage = createAttributedElements({
+            tag:"img",
+            valuesByAttributes:{
+                class:"pointer-events-none",
+                src:button.src,
+                alt:button.alt
+            }
+        })
+
+        fragment.appendChild(sortButton).appendChild(sortButtonImage);
+    }
+
+    sortButtonsBox.appendChild(fragment);
+    return sortButtonsBox;
+}
+
+const toggleHiddenClassForSortButton = (currentButton,nextButton) => {
+    currentButton.classList.add("hidden");
+    nextButton.classList.remove("hidden");
+}
+
+const updateTableBody = usersData =>  {
+    document.getElementById("js-usersTableBody").remove();
+    renderTableBody(usersData);
+};
+
+const sortUsersData = (category, usersData, state) => {
+    const cloneUsersData = [...usersData];
+
+    switch (state) {
+        case "asc":
+            cloneUsersData.sort((a, b) => a[category] - b[category]);
+            updateTableBody(cloneUsersData);
+            break;
+        case "desc":
+            cloneUsersData.sort((a, b) => b[category] - a[category]);
+            updateTableBody(cloneUsersData);
+            break;
+        case "both":
+            updateTableBody(usersData);
+            break;
+
+        default:
```

Comment:
IMO
I thought it would be better to align the default usage with `changeState` as well.
However, the code is working fine and does not need to be fixed. It is a minor IMO.
```suggestion
        default:
             updateTableBody(usersData);
```

Reply:
Thank you for your comment.
I will try to include it in my next Issue!